### PR TITLE
[rom_ext] Generate Format Description

### PR DESCRIPTION
**Pull request by lenary**
This replaces a hand-written and hand-maintained shell script with a
script that uses the information inside `manifest.hjson` to create
the information about the format layout.

It made sense to slightly change how `rom-ext-manifest-generator.py` took its arguments, as this is not a code format, and also to update and clarify the relevant documentation there.

### CODE REVIEW

File: util/rom-ext-manifest-generator.py [hunk 1]
```diff
@@ -52,12 +53,13 @@
   rom-ext-manifest-generator --output-dir:
     Directory where manifest.h will be created.
 
-  rom-ext-manifest-generator --output-headers=[all | c | rust]:
-    Type of headers to be generated:
+  rom-ext-manifest-generator --output-files=[all | c | rust]:
```
Comment (silvestrst) [inline util/rom-ext-manifest-generator.py:56]: nit: `[all | c | rust]` => `[all | c | rust | format]`

File: util/rom-ext-manifest-generator.py [hunk 2]
```diff
@@ -52,12 +53,13 @@
...
+    Type of files to be generated:
 
-    all  - All header files
+    all  - All files listed below
     c    - C header file
-    rust - Rust "header" file
+    rust - Rust module file
+    format - Format description file (plaintext)
```
Comment (silvestrst) [inline util/rom-ext-manifest-generator.py:62]: nit: could you align the `-` signs (pedantic I know :smiley:), makes it slightly nicer to read (I think).

File: util/rom-ext-manifest-generator.py [hunk 3]
```diff
@@ -108,22 +111,20 @@ def generate_defines(fields):
                 regions.append((field['name'], region))
             current_offset_bytes += size_bytes
 
-    return [regions, offsets]
+    return (regions, offsets)
```
Comment (silvestrst) [inline util/rom-ext-manifest-generator.py:114]: Just out of curiosity, why this change?

**Review by silvestrst — APPROVED** [review]
@lenary thanks!

Looks good to me! Additional thanks for factoring out `generate_defines`!

File: util/rom-ext-manifest-generator.py (hunk 3 above)
Comment (lenary) [reply util/rom-ext-manifest-generator.py:114]: I feel it's more idiomatic python to use a tuple when you're returning two things that aren't conceptually a sequence of the same kinds of things - here they're a pair of separate items.
